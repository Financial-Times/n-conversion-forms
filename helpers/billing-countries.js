import { billingCountries } from 'n-common-static-data';

export { billingCountries };
