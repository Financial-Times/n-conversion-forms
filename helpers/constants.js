const printRegions = {
	cemeaV1: 'CEMEA_V1',
	cemeaV2: 'CEMEA_V2',
	apac: 'APAC',
};

const addressLine1AutoComplete = 'address-line1';
const addressLine2AutoComplete = 'address-line2';
const addressLine3AutoComplete = 'address-line3';

module.exports = {
	printRegions,
	addressLine1AutoComplete,
	addressLine2AutoComplete,
	addressLine3AutoComplete,
};
