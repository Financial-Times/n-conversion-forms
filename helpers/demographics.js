const { demographics } = require('n-common-static-data');

export { demographics };
