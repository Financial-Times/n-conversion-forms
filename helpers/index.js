module.exports = {
	'ncf-countries': require('./ncf-countries'),
	'ncf-common-data': require('./ncf-common-data'),
	cemeaV1ISO: require('./supportedCountries').cemeaV1ISO,
	cemeaV2ISO: require('./supportedCountries').cemeaV2ISO,
	apacISO: require('./supportedCountries').apacISO,
	identifyFTShippingZone: require('./supportedCountries')
		.identifyFTShippingZone,
	countriesSupported: require('./supportedCountries').countriesSupported,
	countriesSupportedISO: require('./supportedCountries').countriesSupportedISO,
	supportedPostcodeValidators: require('./supportedPostcodeValidators')
		.supportedPostcodeValidators,
	allSupportedPostcodeExamples: require('./supportedPostcodeExamples')
		.allSupportedPostcodeExamples,
	demographics: require('./demographics').demographics,
	billingCountries: require('./billing-countries').billingCountries,
};
