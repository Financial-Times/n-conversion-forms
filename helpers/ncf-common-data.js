const _get = require('lodash.get');
const COMMON_DATA = require('n-common-static-data');

module.exports = function ({ hash, fn }) {
	const context = Object.assign({}, { [hash.export]: _get(COMMON_DATA, hash.import) }, this);
	return fn(context);
};
