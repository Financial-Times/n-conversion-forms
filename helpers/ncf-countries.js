const { countries } = require('n-common-static-data').billingCountries;

const MINIMUM_TO_GROUP = 50;
const MINIMUM_TO_SHOW_GROUPS = 2;

module.exports = function ({ hash = {}, fn }) {
	const filter = Array.isArray(hash.filterList) ? hash.filterList : false;
	const value = hash.value;

	let data = countries;

	// Only show countries in the filter
	if (filter) {
		data = data.filter(item => filter.includes(item.code));
	}

	// Select the country
	if (value) {
		data = data.map(item => {
			item.selected = item.code === value;
			return item;
		});
	}

	// Group countries
	if (data.length >= MINIMUM_TO_GROUP) {
		data = groupCountries(data);
	}

	const context = Object.assign({ countries: data }, this);
	return fn(context);
};

/**
 * Produce a frequently used option group
 * @param {Array} countries
 * @return {Array}
 */
function groupCountries (countries) {
	const frequentlyUsed = ['GBR', 'USA', 'JPN', 'FRA', 'CAN'];

	const frequentlyUsedCountries = countries
		.filter(item => frequentlyUsed.includes(item.code))
		.map(item => Object.assign({}, item))
		.sort((a, b) => frequentlyUsed.indexOf(a.code) - frequentlyUsed.indexOf(b.code));

	// Not worth showing return standard countries list
	if (frequentlyUsedCountries.length < MINIMUM_TO_SHOW_GROUPS) {
		return countries;
	}

	// If selected country is in frequently used only leave that one selected
	// otherwise the user is shown the lowest selected option
	const selected = frequentlyUsedCountries.find(item => item.selected);
	if (selected) {
		countries.forEach(item => item.selected = false);
	}

	return [
		{ label: 'Frequently Used', countries: frequentlyUsedCountries },
		{ label: 'Alphabetical', countries }
	];
}
