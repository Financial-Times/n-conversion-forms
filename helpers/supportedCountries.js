const supportedCountriesMasterList = {
	cemeaV1: {
		AUT: 'Austria',
		BEL: 'Belgium',
		BGR: 'Bulgaria',
		HRV: 'Croatia',
		CZE: 'Czechia',
		DNK: 'Denmark',
		FIN: 'Finland',
		DEU: 'Germany',
		GRC: 'Greece',
		HUN: 'Hungary',
		ISL: 'Iceland',
		ITA: 'Italy',
		LIE: 'Liechtenstein',
		LTU: 'Lithuania',
		NLD: 'Netherlands',
		NOR: 'Norway',
		POL: 'Poland',
		PRT: 'Portugal',
		QAT: 'Qatar',
		ROU: 'Romania',
		RUS: 'Russian Federation',
		SVK: 'Slovakia',
		SVN: 'Slovenia',
		ZAF: 'South Africa',
		ESP: 'Spain',
		SWE: 'Sweden',
		CHE: 'Switzerland'
	},
	cemeaV2: { FRA: 'France', LUX: 'Luxembourg', MCO: 'Monaco' },
	apac: {
		CHN: 'China',
		HKG: 'Hong Kong',
		IND: 'India',
		IDN: 'Indonesia',
		JPN: 'Japan',
		KOR: 'Korea, Republic of',
		MYS: 'Malaysia',
		PHL: 'Philippines',
		SGP: 'Singapore',
		TWN: 'Taiwan, Province of China',
		THA: 'Thailand'
	},
	GBR: 'United Kingdom',
	USA: 'United States',
	CAN: 'Canada'
};


const cemeaV1ISO = Object.keys(supportedCountriesMasterList.cemeaV1);
const cemeaV2ISO = Object.keys(supportedCountriesMasterList.cemeaV2);
const apacISO = Object.keys(supportedCountriesMasterList.apac);
const countriesSupported = Object.assign({}, ...(function _flatten (o) { return [].concat(...Object.keys(o).map(k => typeof o[k] === 'object' ? _flatten(o[k]) : ({[k]: o[k]})));}(supportedCountriesMasterList)));


module.exports = { cemeaV1ISO, cemeaV2ISO, apacISO, countriesSupported };
