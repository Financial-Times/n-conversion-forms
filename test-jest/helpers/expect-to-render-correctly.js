import ReactDOMServer from 'react-dom/server';

export const expectToRenderCorrectly = {
	toRenderCorrectly(reactComponent, props) {
		// Render the React component with props.
		const ReactComponentRendered = ReactDOMServer.renderToStaticMarkup(
			reactComponent(props)
		);

		// eslint-disable-next-line jest/no-standalone-expect
		expect(ReactComponentRendered).toMatchSnapshot();

		// This point is reached when the above assertions were successful.
		// The test should therefore always pass, that means it needs to be
		// `true` when used normally, and `false` when `.not` was used.
		return {
			pass: !this.isNot,
		};
	},
};
