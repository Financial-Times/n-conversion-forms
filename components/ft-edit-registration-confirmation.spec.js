import { FTEditRegistrationConfirmation } from './index';
import { expectToRenderCorrectly } from '../test-jest/helpers/expect-to-render-correctly';

/***
 * This component has been created for a trial feature for the FT Edit app to be used inside
 * next-subscribe. This can be removed after the trial. The trial is beginning May 2024, please
 * reach out to light-apps channel if we dont reach out first, as to when this can be deleted.
 */

expect.extend(expectToRenderCorrectly);

describe('RegistrationConfirmation', () => {
	it('renders with default props', () => {
		const props = {};

		expect(FTEditRegistrationConfirmation).toRenderCorrectly(props);
	});

	it('renders with a custom email', () => {
		const props = { email: 'test@example.com' };

		expect(FTEditRegistrationConfirmation).toRenderCorrectly(props);
	});

	it('renders an explore hub button always', () => {
		const props = { email: 'test@example.com', hubUrl: 'http://thing-hub' };

		expect(FTEditRegistrationConfirmation).toRenderCorrectly(props);
	});

	it('renders a continue reading button when an articleUrl is provided', () => {
		const props = {
			email: 'test@example.com',
			articleUrl:
				'https://www.ft.com/content/9141eee7-825a-41d7-913d-49e8c12e76db',
		};

		expect(FTEditRegistrationConfirmation).toRenderCorrectly(props);
	});
});
