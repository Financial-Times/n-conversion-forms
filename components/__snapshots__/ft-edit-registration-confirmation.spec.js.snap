// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationConfirmation renders a continue reading button when an articleUrl is provided 1`] = `
<div class="ncf">
  <div class="ncf__center ncf__margin">
    <div class="ncf__icon ncf__icon--tick ncf__icon--large">
    </div>
    <h1 class="ncf__header ncf__header--confirmation">
      Success
    </h1>
    <p class="ncf__FT_Edit_confirmation--message">
      Thank you for registering for FT Edit.
    </p>
    <p class="ncf__FT_Edit_confirmation--message margin-top-x4 ">
      Enjoy 30 days of great journalism. We’ve sent confirmation to test@example.com.
    </p>
  </div>
  <div class="ncf__divider-horizontal">
  </div>
  <div class="ncf__FT_Edit_confirmation--links ncf__margin">
    <div class="ncf__FT_Edit_confirmation--socials">
      <b>
        Breaking news alerts, direct to your lock screen
      </b>
      <p class="ncf__FT_Edit_confirmation--mobile">
        Download our apps or follow us on our socials
      </p>
      <p class="ncf__FT_Edit_confirmation--desktop">
        Find us in the app stores or follow us on our socials
      </p>
      <div class="ncf__FT_Edit_confirmation--badges">
        <div class="o-social-follow"
             aria-label="Follow on social media"
        >
          <a href="https://www.instagram.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--instagram"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-instagram"
          >
            <span class="o-social-follow-icon__label">
              on instagram
            </span>
          </a>
          <a href="https://www.linkedin.com/company/financial-times"
             class="o-social-follow-icon o-social-follow-icon--linkedin"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-linkedin"
          >
            <span class="o-social-follow-icon__label">
              on linkedin
            </span>
          </a>
          <a href="https://twitter.com/FT"
             class="o-social-follow-icon o-social-follow-icon--twitter"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-twitter"
          >
            <span class="o-social-follow-icon__label">
              on twitter
            </span>
          </a>
          <a href="https://www.facebook.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--facebook"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-facebook"
          >
            <span class="o-social-follow-icon__label">
              on facebook
            </span>
          </a>
        </div>
        <div class="ncf__FT_Edit_confirmation--mobile ncf__FT_Edit_confirmation--app-badges"
             aria-label="Download the app"
        >
          <a href="https://apps.apple.com/gb/app/ft-edit-by-the-financial-times/id1574510369"
             class="ncf-icon ncf-icon--apple"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-app-apple-ft-edit"
          >
            <span class="o-social-follow-icon__label">
              on apple
            </span>
          </a>
        </div>
      </div>
    </div>
    <div class="ncf__FT_Edit_confirmation--alerts">
      <b>
        Set up email alerts in
        <span aria-label="myFT"
              class="ncf-icon ncf-icon--myft"
        >
        </span>
      </b>
      <p>
        Choose the content you want to follow by personalising your alerts for the most important topics and additional newsletters
      </p>
      <a href="/myft"
         class="margin-top-x4 ncf__button ncf__button--secondary"
         data-trackable="register-personalise-my-alerts"
      >
        Personalise my alerts
      </a>
    </div>
  </div>
  <a href="/"
     target="_parent"
     class=" ncf__FT_Edit_confirmation--finish ncf__button ncf__button--submit"
     data-trackable="register-finish-head-to-ft-edit-hub"
  >
    Explore our FT Edit hub
  </a>
  <a href="https://www.ft.com/content/9141eee7-825a-41d7-913d-49e8c12e76db"
     target="_parent"
     class=" ncf__FT_Edit_confirmation--finish-article-read ncf__button ncf__button--secondary reduce-bottom-spacing"
     data-trackable="register-finish-head-to-article"
  >
    Continue reading
  </a>
</div>
`;

exports[`RegistrationConfirmation renders an explore hub button always 1`] = `
<div class="ncf">
  <div class="ncf__center ncf__margin">
    <div class="ncf__icon ncf__icon--tick ncf__icon--large">
    </div>
    <h1 class="ncf__header ncf__header--confirmation">
      Success
    </h1>
    <p class="ncf__FT_Edit_confirmation--message">
      Thank you for registering for FT Edit.
    </p>
    <p class="ncf__FT_Edit_confirmation--message margin-top-x4 ">
      Enjoy 30 days of great journalism. We’ve sent confirmation to test@example.com.
    </p>
  </div>
  <div class="ncf__divider-horizontal">
  </div>
  <div class="ncf__FT_Edit_confirmation--links ncf__margin">
    <div class="ncf__FT_Edit_confirmation--socials">
      <b>
        Breaking news alerts, direct to your lock screen
      </b>
      <p class="ncf__FT_Edit_confirmation--mobile">
        Download our apps or follow us on our socials
      </p>
      <p class="ncf__FT_Edit_confirmation--desktop">
        Find us in the app stores or follow us on our socials
      </p>
      <div class="ncf__FT_Edit_confirmation--badges">
        <div class="o-social-follow"
             aria-label="Follow on social media"
        >
          <a href="https://www.instagram.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--instagram"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-instagram"
          >
            <span class="o-social-follow-icon__label">
              on instagram
            </span>
          </a>
          <a href="https://www.linkedin.com/company/financial-times"
             class="o-social-follow-icon o-social-follow-icon--linkedin"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-linkedin"
          >
            <span class="o-social-follow-icon__label">
              on linkedin
            </span>
          </a>
          <a href="https://twitter.com/FT"
             class="o-social-follow-icon o-social-follow-icon--twitter"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-twitter"
          >
            <span class="o-social-follow-icon__label">
              on twitter
            </span>
          </a>
          <a href="https://www.facebook.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--facebook"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-facebook"
          >
            <span class="o-social-follow-icon__label">
              on facebook
            </span>
          </a>
        </div>
        <div class="ncf__FT_Edit_confirmation--mobile ncf__FT_Edit_confirmation--app-badges"
             aria-label="Download the app"
        >
          <a href="https://apps.apple.com/gb/app/ft-edit-by-the-financial-times/id1574510369"
             class="ncf-icon ncf-icon--apple"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-app-apple-ft-edit"
          >
            <span class="o-social-follow-icon__label">
              on apple
            </span>
          </a>
        </div>
      </div>
    </div>
    <div class="ncf__FT_Edit_confirmation--alerts">
      <b>
        Set up email alerts in
        <span aria-label="myFT"
              class="ncf-icon ncf-icon--myft"
        >
        </span>
      </b>
      <p>
        Choose the content you want to follow by personalising your alerts for the most important topics and additional newsletters
      </p>
      <a href="/myft"
         class="margin-top-x4 ncf__button ncf__button--secondary"
         data-trackable="register-personalise-my-alerts"
      >
        Personalise my alerts
      </a>
    </div>
  </div>
  <a href="http://thing-hub"
     target="_parent"
     class=" ncf__FT_Edit_confirmation--finish ncf__button ncf__button--submit"
     data-trackable="register-finish-head-to-ft-edit-hub"
  >
    Explore our FT Edit hub
  </a>
</div>
`;

exports[`RegistrationConfirmation renders with a custom email 1`] = `
<div class="ncf">
  <div class="ncf__center ncf__margin">
    <div class="ncf__icon ncf__icon--tick ncf__icon--large">
    </div>
    <h1 class="ncf__header ncf__header--confirmation">
      Success
    </h1>
    <p class="ncf__FT_Edit_confirmation--message">
      Thank you for registering for FT Edit.
    </p>
    <p class="ncf__FT_Edit_confirmation--message margin-top-x4 ">
      Enjoy 30 days of great journalism. We’ve sent confirmation to test@example.com.
    </p>
  </div>
  <div class="ncf__divider-horizontal">
  </div>
  <div class="ncf__FT_Edit_confirmation--links ncf__margin">
    <div class="ncf__FT_Edit_confirmation--socials">
      <b>
        Breaking news alerts, direct to your lock screen
      </b>
      <p class="ncf__FT_Edit_confirmation--mobile">
        Download our apps or follow us on our socials
      </p>
      <p class="ncf__FT_Edit_confirmation--desktop">
        Find us in the app stores or follow us on our socials
      </p>
      <div class="ncf__FT_Edit_confirmation--badges">
        <div class="o-social-follow"
             aria-label="Follow on social media"
        >
          <a href="https://www.instagram.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--instagram"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-instagram"
          >
            <span class="o-social-follow-icon__label">
              on instagram
            </span>
          </a>
          <a href="https://www.linkedin.com/company/financial-times"
             class="o-social-follow-icon o-social-follow-icon--linkedin"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-linkedin"
          >
            <span class="o-social-follow-icon__label">
              on linkedin
            </span>
          </a>
          <a href="https://twitter.com/FT"
             class="o-social-follow-icon o-social-follow-icon--twitter"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-twitter"
          >
            <span class="o-social-follow-icon__label">
              on twitter
            </span>
          </a>
          <a href="https://www.facebook.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--facebook"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-facebook"
          >
            <span class="o-social-follow-icon__label">
              on facebook
            </span>
          </a>
        </div>
        <div class="ncf__FT_Edit_confirmation--mobile ncf__FT_Edit_confirmation--app-badges"
             aria-label="Download the app"
        >
          <a href="https://apps.apple.com/gb/app/ft-edit-by-the-financial-times/id1574510369"
             class="ncf-icon ncf-icon--apple"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-app-apple-ft-edit"
          >
            <span class="o-social-follow-icon__label">
              on apple
            </span>
          </a>
        </div>
      </div>
    </div>
    <div class="ncf__FT_Edit_confirmation--alerts">
      <b>
        Set up email alerts in
        <span aria-label="myFT"
              class="ncf-icon ncf-icon--myft"
        >
        </span>
      </b>
      <p>
        Choose the content you want to follow by personalising your alerts for the most important topics and additional newsletters
      </p>
      <a href="/myft"
         class="margin-top-x4 ncf__button ncf__button--secondary"
         data-trackable="register-personalise-my-alerts"
      >
        Personalise my alerts
      </a>
    </div>
  </div>
  <a href="/"
     target="_parent"
     class=" ncf__FT_Edit_confirmation--finish ncf__button ncf__button--submit"
     data-trackable="register-finish-head-to-ft-edit-hub"
  >
    Explore our FT Edit hub
  </a>
</div>
`;

exports[`RegistrationConfirmation renders with default props 1`] = `
<div class="ncf">
  <div class="ncf__center ncf__margin">
    <div class="ncf__icon ncf__icon--tick ncf__icon--large">
    </div>
    <h1 class="ncf__header ncf__header--confirmation">
      Success
    </h1>
    <p class="ncf__FT_Edit_confirmation--message">
      Thank you for registering for FT Edit.
    </p>
    <p class="ncf__FT_Edit_confirmation--message margin-top-x4 ">
      Enjoy 30 days of great journalism. We’ve sent confirmation to your email.
    </p>
  </div>
  <div class="ncf__divider-horizontal">
  </div>
  <div class="ncf__FT_Edit_confirmation--links ncf__margin">
    <div class="ncf__FT_Edit_confirmation--socials">
      <b>
        Breaking news alerts, direct to your lock screen
      </b>
      <p class="ncf__FT_Edit_confirmation--mobile">
        Download our apps or follow us on our socials
      </p>
      <p class="ncf__FT_Edit_confirmation--desktop">
        Find us in the app stores or follow us on our socials
      </p>
      <div class="ncf__FT_Edit_confirmation--badges">
        <div class="o-social-follow"
             aria-label="Follow on social media"
        >
          <a href="https://www.instagram.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--instagram"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-instagram"
          >
            <span class="o-social-follow-icon__label">
              on instagram
            </span>
          </a>
          <a href="https://www.linkedin.com/company/financial-times"
             class="o-social-follow-icon o-social-follow-icon--linkedin"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-linkedin"
          >
            <span class="o-social-follow-icon__label">
              on linkedin
            </span>
          </a>
          <a href="https://twitter.com/FT"
             class="o-social-follow-icon o-social-follow-icon--twitter"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-twitter"
          >
            <span class="o-social-follow-icon__label">
              on twitter
            </span>
          </a>
          <a href="https://www.facebook.com/financialtimes"
             class="o-social-follow-icon o-social-follow-icon--facebook"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-social-facebook"
          >
            <span class="o-social-follow-icon__label">
              on facebook
            </span>
          </a>
        </div>
        <div class="ncf__FT_Edit_confirmation--mobile ncf__FT_Edit_confirmation--app-badges"
             aria-label="Download the app"
        >
          <a href="https://apps.apple.com/gb/app/ft-edit-by-the-financial-times/id1574510369"
             class="ncf-icon ncf-icon--apple"
             target="_blank"
             rel="noopener noreferrer"
             data-trackable="register-app-apple-ft-edit"
          >
            <span class="o-social-follow-icon__label">
              on apple
            </span>
          </a>
        </div>
      </div>
    </div>
    <div class="ncf__FT_Edit_confirmation--alerts">
      <b>
        Set up email alerts in
        <span aria-label="myFT"
              class="ncf-icon ncf-icon--myft"
        >
        </span>
      </b>
      <p>
        Choose the content you want to follow by personalising your alerts for the most important topics and additional newsletters
      </p>
      <a href="/myft"
         class="margin-top-x4 ncf__button ncf__button--secondary"
         data-trackable="register-personalise-my-alerts"
      >
        Personalise my alerts
      </a>
    </div>
  </div>
  <a href="/"
     target="_parent"
     class=" ncf__FT_Edit_confirmation--finish ncf__button ncf__button--submit"
     data-trackable="register-finish-head-to-ft-edit-hub"
  >
    Explore our FT Edit hub
  </a>
</div>
`;
